### DISCUSSION TITLE
Use of Amazon S3 bucket as media storage

### DISCUSSION THREAD

**Discussion by HerinneringenOpLinnen** [lunarphp/lunar#1746] [Q&A]
I installed a new Lunar project and got the hub dashboard / admin panel working.
I have an Amazon S3 bucket I want to use for storing all the media that is uploaded through the admin panel. And also serve the images from this same S3 bucket through the Lunar models in my website project.

I configured S3 in config/filesystems.php but when I upload product images via the admin panel the images are still stored on the local disk. I can't seem to find which storage disk is used and how to change that.
Any ideas on how to do this when I want to use S3?

Thanks.

**Comment by glennjacobs** [comment]
See https://spatie.be/docs/laravel-medialibrary/v11/installation-setup

You can set the following in your `.env` file

```
MEDIA_DISK=s3
```

**Reply by HerinneringenOpLinnen** [reply]
I added the MEDIA_DISK and installed the `league/flysystem-aws-s3-v3` package and S3 is used now for storing the media files.
Thanks for the quick response!

**Reply by HerinneringenOpLinnen** [reply]
So S3 is now working properly in the Lunar admin panel. I can upload to S3 and also view them in the browser.

Now I am looking to serve these images from my website project as well. 
In that other project I also configured the S3 disk with `MEDIA_DISK` in `.env`.
I read this documentation: https://spatie.be/docs/laravel-medialibrary/v9/installation-setup#content-adding-a-media-disk
and configured the disk in `config/filesystems.php` and `config/lunar/media.php`

But the image urls still having the local disk structure instead of aws s3.
I already tried reloading config etc.
Any suggestions what I might be missing here?

**Reply by glennjacobs** [reply]
Difficult to tell without seeing any code. 

Have you set `AWS_URL` correctly?

**Reply by HerinneringenOpLinnen** [reply]
After recompiling the frontend project it works now.
The frontend is using Inertia.js and Vue.js so I guess the media config changes for S3 were only available after and not before recompiling assets.
Thanks!
